lib/dictator.ex

```diff
@@ -99,12 +100,19 @@ defmodule Dictator do
 
     params = %{params: conn.params, resource: resource, opts: opts}
 
-    if apply(policy, :can?, [user, action, params]) do
-      conn
-    else
-      unauthorized_handler = unauthorized_handler()
-      opts = unauthorized_handler.init(opts)
-      unauthorized_handler.call(conn, opts)
+    case apply(policy, :can?, [user, action, params]) do
+      true ->
+        conn
+
+      false ->
+        unauthorized_handler = unauthorized_handler()
+        opts = unauthorized_handler.init(opts)
+        unauthorized_handler.call(conn, opts)
+
+      nil ->
```

Comment:
How would you feel about explicitly returning `{:error, :not_found}` instead of `nil`? Feels more explicit to me. cc @naps62 - any particular thoughts?

Reply:
this is the return value of `:can?/3`. That function has a boolean response (enforced by the question mark, expanding that API would break the semantics)

I didn't notice at first that this is what the PR was doing: allowing `:can?/3` to return `nil`. But that seems like a no-go to me

It also makes me wonder that this might be outside of the scope of dictator?
we could maybe allow an optional `exists?` callback to the policy, to be called before `can?`. But it would have to be opt-in, and I'm not sure how to handle that correctly

Reply:
My original issue with verifying this outside Dictator and before the Dictator plug was called, was information could be leaked about whether or not an object exists to someone who we didn't check yet if they have the authorization to access that information. 

But thinking a bit about what you're saying, maybe the approach was not the most correct one, since indeed it seems to fall out of the current responsibilities of the can? function. As for the scope of dictator, an optional exists? to me would look ok, just because of the reason above about leaking information when verifying authorizations.